cli/mender-artifact/artifacts.go

```diff
@@ -330,13 +330,13 @@ func repack(comp artifact.Compressor, ua *unpackedArtifact, to io.Writer, key []
 		aWriter = awriter.NewWriterSigned(to, comp, artifact.NewSigner(key))
 	}
 
-	// for rootfs-images: Update rootfs_image_checksum provide if there is one.
-	_, hasChecksumProvide := ua.writeArgs.TypeInfoV3.ArtifactProvides["rootfs_image_checksum"]
+	// for rootfs-images: Update rootfs-image.checksum provide if there is one.
+	_, hasChecksumProvide := ua.writeArgs.TypeInfoV3.ArtifactProvides["rootfs-image.checksum"]
```

Comment:
This is just modifying existing artifacts, so we should probably check for both here, and pass the correct legacy flag below.

Reply:
Done here: https://github.com/mendersoftware/mender-artifact/pull/291/commits/d0c71255610606ded4c1d8d5bf176a040baee2a8